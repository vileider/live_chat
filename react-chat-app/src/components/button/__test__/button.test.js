import React from 'react';
import ReactDOM from 'react-dom';
import Button from '../button';
import { render, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';

import renderer from "react-test-renderer";

import switchTheme  from "../../Chat_functionts/switchtheme";

afterEach(() => {
    cleanup();
});
//clean up after every single test to clean up render

it("renders without crashing", ()=> {
    const div=document.createElement("div");
    ReactDOM.render(<Button></Button>, div)
})

it("renders button correctly", () => {
    const {getByTestId} = render(<Button label="x"></Button>)
    expect(getByTestId('button')).toHaveTextContent("x")
})

it("matches snapchot 1", () => {
    const tree = renderer.create(<Button label="xx"></Button>).toJSON();
    expect(tree).toMatchSnapshot();
})

it("matches snapchot 2", () => {
    const tree = renderer.create(<Button label="x"></Button>).toJSON();
    expect(tree).toMatchSnapshot();
})

test('the best flavor is grapefruit', () => {
    expect(switchTheme(true)).toBe("dark_mode");
  });