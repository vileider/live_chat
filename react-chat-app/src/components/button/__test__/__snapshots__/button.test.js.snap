// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapchot 1`] = `
<div
  className="button-style"
  data-testid="button"
>
  Hi 
  xx
</div>
`;

exports[`matches snapchot 1 1`] = `
<div
  className="button-style"
  data-testid="button"
>
  Hi 
  xx
</div>
`;

exports[`matches snapchot 2 1`] = `
<div
  className="button-style"
  data-testid="button"
>
  Hi 
  x
</div>
`;
